docs/guides/component-testing/component-framework-configuration.mdx

```diff
@@ -14,29 +14,54 @@ guide for a list of supported development servers and how they are configured.
 Below are more advanced configuration options you can customize to fit your
 project.
 
+## Custom Index File
+
+By default, Cypress renders your components into an HTML file located at
+`cypress/support/component-index.html`.
+
+The index file allows you to add in global assets, such as styles, fonts, and
+external scripts.
+
+You can provide an alternative path to the file using the `indexHtmlFile` option
+in the [component config](/guides/references/configuration#component) options:
+
+```js
+{
+  component: {
+    devServer,
+    indexHtmlFile: '/custom/path/to/component-index.html'
+  }
+}
+```
+
 ## Custom Dev Server
 
 A custom function can be passed into the `devServer` option, which allows the
-use of other dev servers not provided by Cypress out of the box. These can be
-from the Cypress community, preview builds not included with the app, or a
-custom one you create.
+use of build systems not provided by Cypress out of the box. These can be from
+the Cypress community, preview builds not included with the app, or a custom one
+you create.
 
-The function's signature takes in a
-[Cypress Configuration](/guides/references/configuration) object as its only
-parameter and returns either an instance of a `devServer` or a promise that
-resolves to a `devServer` instance.
+The function's signature takes in a object with the following properties as its
+only parameter and needs to resolve an object containing the port of your dev
+server.
+
+```ts
+interface DevServerOptions {
+  specs: Cypress.Spec[]
+  cypressConfig: Cypress.PluginConfigOptions
+  devServerEvents: NodeJS.EventEmitter
+}
+```
 
 :::cypress-config-example
 
 ```ts
 {
   component: {
-    devServer(cypressConfig: CypressConfiguration) {
-      // return devServer instance or a promise that resolves to
-      // a dev server here
+    async devServer({specs, cypressConfig, devServerEvents}: DevServerOptions) {
+      const port = await startDevServer(specs, cypressConfig, devServerEvents)
       return {
-        port: 1234,
-        close: () => {},
+        port: port
```

Comment:
Should we include `close` here, maybe

```js
return {
  port,
  close: () => { /* optional code to run when the dev server is closed */ }
}
```
?

Reply:
I skipped it since in the webpack dev-server it was mentioned it's only for unit testing: https://github.com/cypress-io/cypress/blob/develop/npm/webpack-dev-server/src/devServer.ts#L75
I'm not sure if this is stable enough to be used as a public api?

Reply:
Hmm, close seems required in the type-definitions, so I'll add it after all.